lib/http.js

```diff
@@ -0,0 +1,45 @@
+import { getLatestMetrics, followMetrics } from '../lib/metrics.js'
+import http from 'node:http'
+import { once } from 'node:events'
+
+const asyncHandler = async (req, res) => {
+  const url = new URL(req.url, 'http://localhost/')
+
+  if (url.pathname === '/metrics') {
```

Comment:
Since we want to add more endpoints soon, in order to avoid code churn, I am proposing to extract this handler into a standalone function, even if it feels like premature optimisation right now.